include/seqan3/alignment/pairwise/policy/alignment_matrix_policy.hpp

```diff
@@ -0,0 +1,146 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::detail::alignment_matrix_policy.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/alignment/band/static_band.hpp>
+#include <seqan3/alignment/pairwise/detail/alignment_algorithm_cache.hpp>
+#include <seqan3/core/type_traits/basic.hpp>
+#include <seqan3/range/views/slice.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief Manages the alignment and score matrix.
+ *
+ */
+template <typename alignment_algorithm_t, typename score_matrix_type, typename trace_matrix_type>
+class alignment_matrix_policy
```

Comment:
Then it would be called alignment_matrix_policy_base suggesting that this is a base for another policy, which it is not. We wanted to have a subfolder depth of at most 2, which means the policy folder needs to be dropped and instead all policies go into the detail directory of pairwise. Therefore, keeping policy in the naming is essential in my POV.

Reply:
`alignment_matrix_base_policy` is not much better I guess. It then suggests that it is a policy for the base class, which it is not.

Reply:
> Then it would be called alignment_matrix_policy_base suggesting that this is a base for another policy

It's not? :eyes: 
Then what is this CRTP for?

Reply:
> It's not? 👀

No it is not! It is a policy for the alignment algorithm class. Not a base class for a policy for the alignment algorithm class. Hence, policy_base suggests something different. It is a policy and it was implemented by means of CRTP design pattern.